### ISSUE TITLE
retry close() on EINTR?

### ISSUE THREAD

**Issue by tkhattra** [Tarsnap/libcperciva#229]
should close() be retried on EINTR in util/daemonize.c and util/entropy.c?

freebsd close(2) man page says: "In case of any error except EBADF, the supplied file descriptor is de-allocated and therefore is no longer valid."

linux close(2) man page also suggests not retrying close() in case of any error (except EINTR on HP-UX).

in a multi-threaded program, retrying close() may end up closing another thread's file descriptor (this is mentioned in linux close(2) man page too).

**Comment by cperciva** [comment]
Yeah, we should probably change these.  It's complicated since making close(2) work properly is very difficult (https://www.daemonology.net/blog/2011-12-17-POSIX-close-is-broken.html) although it is possible (https://twitter.com/cperciva/status/1141852451756105729).

I'm starting to think that I should just bite the bullet and implement the horrible-but-standards-compliant solution so I can stop worrying about this.

**Comment by tkhattra** [comment]
wow, you already considered this many moons ago and solved it too - kudos!

**Comment by cperciva** [comment]
Well... solved it in theory, at least. ;-)
